packages/myst-cli/src/build/html/index.ts

```diff
@@ -141,25 +145,27 @@ export async function buildHtml(session: ISession, opts: StartOptions) {
 
   // Fetch all HTML pages and assets by the template
   await Promise.all(
-    routes.map(async (route) => {
-      const resp = await session.fetch(route.url);
-      if (!resp.ok) {
-        session.log.error(`Error fetching ${route.url}`);
-        return;
-      }
-      if (route.binary && resp.body) {
-        await new Promise<void>((resolve) => {
-          const filename = path.join(htmlDir, route.path);
-          if (!fs.existsSync(filename)) fs.mkdirSync(path.dirname(filename), { recursive: true });
-          const fileWriteStream = fs.createWriteStream(filename);
-          resp.body!.pipe(fileWriteStream);
-          fileWriteStream.on('finish', resolve);
-        });
-      } else {
-        const content = await resp.text();
-        writeFileToFolder(path.join(htmlDir, route.path), content);
-      }
-    }),
+    routes.map(async (route) =>
+      limitConnections(async () => {
+        const resp = await fetchWithRetry(session, route.url);
```

Comment:
Main changes here, limiting connections to 5 and having a retry with exponential backoff.